src/errors/FixingNonResourceError.ts

```diff
@@ -0,0 +1,7 @@
+export class FixingNonResourceError extends Error {
+  constructor(public resourceType: string, public id: string) {
+    super(
+      `Instance ${id} of type ${resourceType} is not an Instance of a Resource. Only Instances of Resources may be assigned to other Instances.`
```

Comment:
Should we add "or profile of a Resource" just to be more clear that it is allowed?